node/src/p2p/header_session.rs

```diff
@@ -137,10 +135,38 @@ impl HeaderSession {
     }
 }
 
+fn sort_header_spans_into_ranges(
```

Comment:
Add a bit of documentation here. For examples that this input `[1, 2], [7, 8], [3, 4]` will give this result `[1, 2, 3, 4], [7, 8]`.

Also what do you think about this name? `sort_and_flatten_headers`

Reply:
I like the suggestion more than the current version, changed :P 
Touch worried about the canonical interpretation of `flatten` as something that removes one level of array nesting, but I think we're close enough.

Added few words of documentation